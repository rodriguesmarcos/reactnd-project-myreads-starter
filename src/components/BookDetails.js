import React from 'react';
import PropTypes from 'prop-types';
import { FoldingCube } from 'better-react-spinkit';
import { Route } from 'react-router-dom';
import BookShelfChanger from './BookShelfChanger';

const BookDetails = ({match, getViewableBook, onChangeShelf, shelves}) => {

  const { bookId } = match.params;
  const book = getViewableBook(bookId);

  if (!book.id)
    return <FoldingCube size={75} color="#2e7c31" className="loading-spinner" />;

  const bookImage = book.imageLinks && book.imageLinks.smallThumbnail ? book.imageLinks.smallThumbnail : '';    

  return (
    <div className="book-details">
      <div className="book-details-cover">
        <div className="book-top">

          <div className="book-cover" style={{
            width: 128,
            height: 193,
            backgroundImage: `url(${bookImage})`
          }}>
          </div>

          <div className="book-shelf-changer">
            <Route render={(props) => (
              <BookShelfChanger
                {...props}
                book={book}
                shelves={shelves}
                onChangeShelf={onChangeShelf}
              />
            )}/>
          </div>

        </div>
      </div>

      <div>
        <div className="book-details-header">
          <h2 className="book-details-title">{book.title}</h2>
          {book.subtitle && (<h3 className="book-details-subtitle">{book.subtitle}</h3>)}
          {Array.isArray(book.authors) && (<div className="book-details-authors">{book.authors.join(', ')}</div>)}
        </div>

        {book.description && (
          <div className="book-details-content">
            {book.description}
          </div>
        )}

        <div className="book-details-footer">
          <h4 className="book-details-more">More Details</h4>
          <ul>
            {book.pageCount && (<li>Pages: {book.pageCount}</li>)}
            {book.language && (<li>Language: {book.language}</li>)}
            {book.publishedDate && (<li>Published Date: {book.publishedDate}</li>)}
            {book.publisher && (<li>Publisher: {book.publisher}</li>)}
          </ul>
        </div>
      </div>


    </div>
  )
};

BookDetails.propTypes = {
  match: PropTypes.object.isRequired,
  getViewableBook: PropTypes.func.isRequired,
  onChangeShelf: PropTypes.func.isRequired,
}

export default BookDetails;
