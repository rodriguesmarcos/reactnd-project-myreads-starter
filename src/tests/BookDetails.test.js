import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookDetails from '../components/BookDetails.js';
import books from './fixtures/books';

describe('<BookDetails />', () => {

  const minProps = {
    match: { params: { bookId: books[0].id }},
    getViewableBook: () => (books[0]),
    onChangeShelf: () => {},
  }

  test('renders without exploding', () => {
    const wrapper = shallow(<BookDetails {...minProps}/>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
