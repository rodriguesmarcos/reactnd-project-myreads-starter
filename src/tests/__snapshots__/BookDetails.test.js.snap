exports[`<BookDetails /> renders without exploding 1`] = `
<div
  className="book-details">
  <div
    className="book-details-cover">
    <div
      className="book-top">
      <div
        className="book-cover"
        style={
          Object {
            "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
            "height": 193,
            "width": 128,
          }
        } />
      <div
        className="book-shelf-changer">
        <Route
          render={[Function]} />
      </div>
    </div>
  </div>
  <div>
    <div
      className="book-details-header">
      <h2
        className="book-details-title">
        The Linux Command Line
      </h2>
      <h3
        className="book-details-subtitle">
        A Complete Introduction
      </h3>
      <div
        className="book-details-authors">
        William E. Shotts, Jr.
      </div>
    </div>
    <div
      className="book-details-content">
      You\'ve experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you\'ll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book\'s short, easily-digestible chapters, you\'ll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \"shell shock,\" you\'ll find that the command line is a natural and expressive way to communicate with your computer. Just don\'t be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation\'s \"Evolution of a SysAdmin\"
    </div>
    <div
      className="book-details-footer">
      <h4
        className="book-details-more">
        More Details
      </h4>
      <ul>
        <li>
          Pages: 
          480
        </li>
        <li>
          Language: 
          en
        </li>
        <li>
          Published Date: 
          2012
        </li>
        <li>
          Publisher: 
          No Starch Press
        </li>
      </ul>
    </div>
  </div>
</div>
`;
